src/hash/rpo/mod.rs

```diff
@@ -345,18 +358,65 @@ impl Rpo256 {
     pub fn apply_round(state: &mut [Felt; STATE_WIDTH], round: usize) {
         // apply first half of RPO round
         Self::apply_mds(state);
-        Self::add_constants(state, &ARK1[round]);
-        Self::apply_sbox(state);
+        if !Self::optimized_add_constants_and_apply_sbox(state, &ARK1[round]) {
+            Self::add_constants(state, &ARK1[round]);
+            Self::apply_sbox(state);
+        }
```

Comment:
A small nit: I wonder if we should replace this with just:
```
Self::add_constants_and_apply_sbox(state, &ARK1[round]);
```
And then we can have feature-specific implementations of these functions:
```Rust
#[inline(always)]
#[cfg(not(feature = "arch-arm64-sve"))]
fn add_constants_and_apply_sbox(state: &mut [Felt; STATE_WIDTH], ark: &[Felt; STATE_WIDTH]) {
    Self::add_constants(state, &ARK1[round]);
    Self::apply_sbox(state);
}

#[inline(always)]
#[cfg(feature = "arch-arm64-sve")]
fn add_constants_and_apply_sbox(state: &mut [Felt; STATE_WIDTH], ark: &[Felt; STATE_WIDTH]) {
    unsafe {
        sve::add_constants_and_apply_sbox(state.as_mut_ptr() as *mut u64, ark.as_ptr() as *const u64)
    }
}
```
Would require putting C version of `add_constants_and_apply_sbox` into an `sve` module, but I think this might be cleaner.

Reply:
We can merge Rust-native `add_constants` and `apply_sbox` functions, but it won't change the fact that the `optimized_add_constants_and_apply_sbox` might not run. We need a fallback for times when new SVE implementations pop up and change vector sizes.

Reply:
Ah - I missed this part. Let's keep as is.